enzyme/Enzyme/MLIR/Interfaces/GradientUtilsReverse.cpp

```diff
@@ -72,10 +72,34 @@ Type mlir::enzyme::MGradientUtilsReverse::getCacheType(Type t) {
   return cacheType;
 }
 
+void MGradientUtilsReverse::registerCacheCreatorHook(
+    std::function<std::pair<Value, Value>(Type)> hook) {
+  if (hook != nullptr) {
+    cacheCreatorHook.push_back(hook);
+  }
+}
+
+void MGradientUtilsReverse::deregisterCacheCreatorHook(
+    std::function<std::pair<Value, Value>(Type)> hook) {
```

Comment:
It may make sense to template the closure parameter in order to avoid creating a `std::function` object. `template <typename Fun>`